src/guides/workflows/add_header.md

```diff
@@ -0,0 +1,288 @@
+# Add Header & Send Request Workflow
+
+This guide will assist you in building a [Passive Workflow](/reference/workflows/passive.md) that will add a header to an in-scope request and resend the request with it.
+
+## Nodes and Connections
+
+For this Workflow, the overall Node layout will be:
+
+<img alt="Nodes used and their connections." src="/_images/nodes_adding_header.png" center>
+
+- The `On intercept request` will output `$on_intercept_request.request` which represents the request that passed through the Caido proxy.
+- The request will be sent to the `In Scope` Node. This will check if the request is within your current scope.
+- If the request is within scope it will be passed to the `JavaScript` Node. If it is not - the Workflow be exited.
+- Once the request has been processed by the script in the `JavaScript Node`, the Workflow will come to an end.
+
+## Typing
+
+When a JavaScript Node is executed inside a Workflow, the `run` function will be triggered.
+
+The JSDoc comments note the types of the parameters and return value.
+
+::: tip 1.
+
+``` js
+/**
+ * @param {HttpInput} input
+ * @param {SDK} sdk
+ * @returns {MaybePromise<Data | undefined>}
+ */
+```
+
+- This function will take the `input` and `sdk` parameters. The `SDK` object is an interface to the Caido backend.
+- The `input` object is a `HttpInput` object, which represents a request and its associated reponse.
+- The type alias of `{MaybePromise<Data | undefined>}` handles both synchhronous and asynchronous values. It will either return a `byte` value or return `undefined`.
```

Comment:
Two h.
Multiple values can be Data, do we have doc we can link on that?

Reply:
Removed an 'h'. Doing these edits from my phone (hence all the commits).

I'll link/create in the morning.

Reply:
Had to get on the desktop anyways actually. Added link.